src/tokenserver/auth/browserid.rs

```diff
@@ -0,0 +1,479 @@
+use actix_web::{web, Error};
+use async_trait::async_trait;
+use reqwest::{blocking::Client as ReqwestClient, StatusCode};
+use serde::{de::Deserializer, Deserialize, Serialize};
+
+use super::VerifyToken;
+use crate::tokenserver::{
+    error::{ErrorLocation, TokenserverError},
+    settings::Settings,
+};
+
+use core::time::Duration;
+use std::convert::TryFrom;
+
+/// The information extracted from a valid BrowserID assertion.
+#[derive(Clone, Debug, Default, Deserialize, PartialEq)]
+pub struct VerifyOutput {
+    pub device_id: Option<String>,
+    pub email: String,
+    pub generation: Option<i64>,
+    pub keys_changed_at: Option<i64>,
+}
+
+/// The verifier used to verify BrowserID assertions.
+#[derive(Clone)]
+pub struct RemoteVerifier {
+    audience: String,
+    issuer: String,
+    fxa_verifier_url: String,
+    // Note that we do not need to use an Arc here, since Reqwest's blocking client uses an Arc
+    // internally
+    request_client: ReqwestClient,
```

Comment:
If `request::blocking::Client` didn't use an Arc internally, we'd need to use one here to maintain keep-alive connections between threads